Added nl language

berendjan opened the pull request.

kevin-bates commented on notebook/i18n/README.md:
```diff
@@ -53,6 +53,11 @@ pybabel extract -F babel_nbjs.cfg -o nbjs.pot --no-wrap --project Jupyter .
+[GitHub Repository Link](https://github.com/berendjan/Jupyter-Notebook-Azure-Translator.git),
+however this requires an Azure account which provides a free translations API up to 2M characters, this
+is more than enough for the translation of the 3 `.pot` files.
+
```
This paragraph feels a little odd to me given that it minimally requires a free "trial" account.  I agree that its useful to know about, I just don't know if this kind of thing should appear in the README.

berendjan replied:
Please check the below commit, which notes is is only a possible starting point. Is that better?

kevin-bates replied:
The update looks great -thank you.

berendjan commented:
I noticed that the `.po` template files `.pot` are out of sync, the lines to which it refers are often incorrect which leads to not translated text. I'm unsure how to generate new `.pot` files. I presume this is a problem for all translations (French (fr_FR) and Chinees (zh-CN)). 

I raised an [issue #5356](https://github.com/jupyter/notebook/issues/5356)

berendjan commented:
Furthermore, we need more volunteer language checkers to iterate the translations.
Jupyter Notebook retrieves the language from the `LANG` variable in the terminal.
You can set is for the NL language by `export LANG="nl_NL"` for this example in case you have your system settings to English.

berendjan commented:
> @berendjan - thank you for your contribution - Dutch localization!
> 
> Having not "witnessed" the previous localization additions, it's not entirely clear to me what else is necessary for this. On the surface, everything looks good, but I'd prefer another set of eyes on this particular review.
> 
> One reservation I have is that these translations were performed via the [Translator Text API](https://azure.microsoft.com/en-us/services/cognitive-services/translator-text-api/). In my experiences with i18n on other projects, this task was typically performed by humans, often with multiple iterations, due to contextual misunderstandings, etc. Assuming you speak/read Dutch, how well do you feel the translations are in preserving the original intent of the given message? It would be good to minimize the number of fixups required because the machine didn't quite interpret intent correctly.

@kevin-bates I checked the translations and changed them where I found necessary, but you are absolutely right and probably iterations are necessary. I changed the commit to a [WIP] and gave some instructions above, I also noticed that the `.pot` and `.po` files were out of sync and raised [issue #5356](https://github.com/jupyter/notebook/issues/5356).

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbui.po:
```diff
@@ -0,0 +1,746 @@
+
+#: notebook/templates/tree.html:32
+msgid "Select items to perform actions on them."
+msgstr "Selecteer items om acties op deze acties uit te voeren."
```
This is not dutch.
"Selecteer items om acties op uit te voeren."

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbui.po:
```diff
@@ -0,0 +1,746 @@
+
+#: notebook/templates/tree.html:35
+msgid "Duplicate"
+msgstr "Dubbele"
```
"Kopieer"
or
"Dupliceer"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbui.po:
```diff
@@ -0,0 +1,746 @@
+
+#: notebook/templates/tree.html:37
+msgid "Move selected"
+msgstr "Geselecteerd verplaatsen"
```
"Verplaats geselecteerde"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbui.po:
```diff
@@ -0,0 +1,746 @@
+
+#: notebook/templates/tree.html:40
+msgid "View selected"
+msgstr "Weergave geselecteerd"
```
"Geef geselecteerde weer"
and the other "* selected" or wrong as well.

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbui.po:
```diff
@@ -0,0 +1,746 @@
+
+#: notebook/templates/edit.html:63
+msgid "Sublime Text"
+msgstr "Sublieme tekst"
```
Shouldn't this be kept in English (it's a name?)?

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbui.po:
```diff
@@ -0,0 +1,746 @@
+
+#: notebook/templates/login.html:39
+msgid "No login available, you shouldn't be seeing this page."
+msgstr "Geen login beschikbaar, zou u deze pagina niet moeten zien."
```
"Geen login beschikbaar, u zou deze pagina niet moeten zien."

toonijn commented on notebook/i18n/nl/LC_MESSAGES/notebook.po:
```diff
@@ -0,0 +1,605 @@
+
+#: notebook/notebookapp.py:402
+msgid "Currently running servers:"
+msgstr "Momenteel servers draaien:"
```
"Momenteel draaiende servers:"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.po:
```diff
@@ -0,0 +1,2116 @@
+
+#: notebook/static/notebook/js/actions.js:106
+msgid "confirm restart kernel"
+msgstr "opnieuw opstartenkernel bevestigen"
```
"bevestig herstart van kernel"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.po:
```diff
@@ -0,0 +1,2116 @@
+
+#: notebook/static/notebook/js/actions.js:152
+msgid "run cell, select below"
+msgstr "cel uitvoeren, selecteer hieronder"
```
"... eronder"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.po:
```diff
@@ -0,0 +1,2116 @@
+#: notebook/static/notebook/js/actions.js:373
+#: notebook/static/notebook/js/actions.js:374
+msgid "change cell to markdown"
+msgstr "cel wijzigen om af te markeren"
```
"cel wijzigen naar markdown"
double check for other occurrences of 'markdown'.

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.po:
```diff
@@ -0,0 +1,2116 @@
+
+#: notebook/static/notebook/js/actions.js:781
+msgid "scroll cell top"
+msgstr "schuifcelboven"
```
"scroll cell *" are wrongly translated

kevin-bates commented:
@toonijn - thank you for the great review.  I'd like to start down the path of putting together a 6.1.0 release and would love to have this included.  @berendjan - would it be possible to respond to Toon's review comments fairly soon?

berendjan commented:
@toonijn Thank you for your input, it's much appreciated. I have to agree that 'notitieblok' which is a notepad in English sounds weird. I changed everything to notebook as I presume people who come in contact with Jupyter Notebook will be more familiar with the English name even when their system settings are Dutch.

If you have time, please look at the changes and comment away! I'll try to address them asap.

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Geen"
+         ],
+         "No checkpoints": [
+            "Geen controleposten"
```
Maybe also call it a "checkpoint"?

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Alle cellen opnieuw starten en uitvoeren"
+         ],
+         "Restart and Clear All Outputs": [
+            "Alle uitvoer opnieuw starten en wissen"
```
Wrongly translated
Also "herstarten" instead of "opnieuw starten"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Alle uitvoer opnieuw starten en wissen"
+         ],
+         "Restart": [
+            "Opnieuw starten"
```
"Herstarten"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Wilt u de huidige kernel opnieuw opstarten?  Alle variabelen gaan verloren."
+         ],
+         "Shutdown kernel?": [
+            "Shutdown kernel?"
```
"Sluit de kernel af?"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Het notebookbestand op de schijf is gewijzigd sinds de laatste keer dat het geopend of opgeslagen is. Wilt u het bestand op schijf overschrijven met de versie die hier wordt geopend, of de versie op schijf laden (de pagina opnieuw laden) ?"
+         ],
+         "Notebook validation failed": [
+            "Laptopvalidatie is mislukt"
```
Funny (and wrong).

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Opdrachtmodus"
+         ],
+         "Kernel Created": [
+            "Kernel gemaakt"
```
"aangemaakt"?

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "klik om opnieuw verbinding te maken"
+         ],
+         "Restarting kernel": [
+            "Kernel opnieuw opstarten"
```
"herstarten"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Probeer nu opnieuw te starten"
+         ],
+         "The kernel has died, and the automatic restart has failed. It is possible the kernel cannot be restarted. If you are not able to restart the kernel, you will still be able to save the notebook, but running code will no longer work until the notebook is reopened.": [
+            "De kernel is gestopt en de automatische herstart is mislukt. Het is mogelijk dat de kernel niet opnieuw kan worden gestart. Als u de kernel niet opnieuw starten, u het notebook nog steeds opslaan, maar werkt de code niet meer totdat het notebook opnieuw is geopend."
```
"Als u de kernel niet opnieuw **start**, **kan** u het notebook nog steeds opslaan, maar "

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Het notebook wordt elke %d sec opgeslagen."
+         ],
+         "Trusted": [
+            "Vertrouwde"
```
"Vertrouwd"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "ga een woord naar rechts"
+         ],
+         "delete word before": [
+            "woord verwijderen voordat"
```
"woord ervoor verwijderen"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "woord verwijderen voordat"
+         ],
+         "delete word after": [
+            "woord verwijderen na"
```
"woord erna verwijderen"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "woord verwijderen na"
+         ],
+         "code completion or indent": [
+            "code voltooiing of inspringing"
```
"aanvulling" instead of "voltooiing"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Meer dan 100 matches, afbreken"
+         ],
+         "Use regex (JavaScript regex syntax)": [
+            "Regex (syntaxis van JavaScript-regex) gebruiken"
```
"Regex gebruiken (JavaScript regex syntax)"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Schrijf hier in LaTeX of ander format, voor gebruik met nbconvert. Het wordt niet weergegeven in het notebook. Bij het doorlopen van nbconvert wordt de inhoud van een Raw Cell ongewijzigd aan de uitvoer toegevoegd."
+         ],
+         "Grow the tooltip vertically (press shift-tab twice)": [
+            "De tooltip verticaal laten groeien (druk twee maal op shift-tab)"
```
"twee keer"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "De tooltip verticaal laten groeien (druk twee maal op shift-tab)"
+         ],
+         "show the current docstring in pager (press shift-tab 4 times)": [
+            "de huidige docstring weergeven in pager (druk op shift-tab 4 keer)"
```
"druk vier keer op shift-tab"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "De tooltip blijft 10 seconden hangen terwijl u typt"
+         ],
+         "Welcome to the Notebook Tour": [
+            "Welkom bij de Notebook Tour"
```
"Tour" -> "rondleiding"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Welkom bij de Notebook Tour"
+         ],
+         "You can use the left and right arrow keys to go backwards and forwards.": [
+            "U de pijltoetsen links en rechts gebruiken om heen en weer te gaan."
```
"U kan de pijltjestoetsen ..."

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Op dit moment bent u in de opdrachtmodus en zijn er veel sneltoetsen beschikbaar. In deze modus wordt geen pictogram weergegeven in het indicatorgebied."
+         ],
+         "Pressing <code>Enter</code> or clicking in the input text area of the cell switches to Edit Mode.": [
+            "Als u op <code>Enter </code> of klikt in het invoertekstgebied van de cel, schakelt u over naar de bewerkingsmodus."
```
"Als u op <code>Enter</code> **drukt** of ..."

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "De kernel onderbreken"
+         ],
+         "To cancel a computation in progress, you can click here.": [
+            "Als u een actieve berekening wilt annuleren, u hier klikken."
```
"..., **kan** u hier klikken"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+         "Messages in response to user actions (Save, Interrupt, etc.) appear here.": [
+            "Berichten in reactie op acties van gebruikers (Opslaan, Onderbreken, enz.) worden hier weergegeven."
+         ],
+         "End of Tour": [
+            "Einde van de Tour"
+         ],
+         "This concludes the Jupyter Notebook User Interface Tour.": [
+            "Dit concludeert de Jupyter Notebook User Interface Tour."
+         ],
```
"Tour" -> "rondleiding"

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+         ],
+         "Enter a new destination directory path for this item:": [
+            "Voer een nieuw bestemmingsmap-pad in voor dit item:",
+            "Voer een nieuw bestemmingsmap-pad in voor deze %d-items:"
+         ],
+         "Move an Item": [
+            "Een item verplaatsen",
+            "%d-objecten verplaatsen"
```
Is the minus (-) necessary?

toonijn commented on notebook/i18n/nl/LC_MESSAGES/nbjs.json:
```diff
@@ -0,0 +1,1353 @@
+            "Verplaatsen mislukt"
+         ],
+         "Are you sure you want to permanently delete: \"%s\"?": [
+            "Weet u zeker dat u het volgende definitief wilt verwijderen: \"%s\"?",
+            "Weet u zeker dat u de geselecteerde %d-bestanden of mappen permanent wilt verwijderen?"
```
Not the same sentence.
Is the minus necessary?

toonijn commented on notebook/i18n/nl/LC_MESSAGES/notebook.po:
```diff
@@ -0,0 +1,607 @@
+"Alternatively use `%s` when working on the notebook's Javascript and LESS"
+msgstr ""
+"U ook '%s' gebruiken bij het werken aan de Javascript van de notebook en "
+"MINDER"
```
"MINDER " should be less

toonijn commented on notebook/i18n/nl/LC_MESSAGES/notebook.po:
```diff
@@ -0,0 +1,607 @@
+"Supply overrides for the tornado.web.Application that the Jupyter notebook "
+"uses."
+msgstr ""
+"Levering overschrijft voor de tornado.web.Application die de Jupyter "
```
Weird sentence

berendjan replied:
"Supply overrides voor de tornado.web.Application die de Jupyter"
" notebook gebruikt." is this better?
I'm not sure what the sentence is referring too in this case. The line numbering seems to be out of sync.

toonijn replied:
I have no idea as well.

If I search for it in the code I get:
https://github.com/jupyter/notebook/blob/b4a2ad990a49479965b681ddc89ab0a9e993a2fc/notebook/notebookapp.py#L1001

From cotext it seems like a user can change the default settings for the tornado.web.Application by supplying some extra config. So maybe a translation like:
"Geef extra instellingen voor de 'tornado.web.Application' die gebruikt wordt door de Jupyter notebook."
could work.

berendjan replied:
agreed.

toonijn commented:
Looks good to me!

kevin-bates commented:
Fantastic - thank you both @berendjan and @toonijn!  

@toonijn - could you please select approve on your review, then I think we'll merge in a day or so to allow others to chime in one last time.

kevin-bates commented:
Thanks @toonijn.  

@berendjan - if you're all set, please update the title and we'll take the next step.

berendjan commented:
@toonijn @kevin-bates Thanks to you both, I removed [WIP] from the title.
